src/certificates.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utilities for generating certificates."""
+
+import logging
+from datetime import datetime, timedelta, timezone
+from typing import List, Optional
+
+from cryptography import x509
+from cryptography.hazmat._oid import ExtensionOID
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+
+logger = logging.getLogger(__name__)
+
+
+def generate_private_key(
+    key_size: int = 2048,
+    public_exponent: int = 65537,
+) -> str:
+    """Generate a private key.
+
+    Args:
+        key_size (int): Key size in bytes
+        public_exponent: Public exponent.
+
+    Returns:
+        str: Private Key
+    """
+    private_key = rsa.generate_private_key(
+        public_exponent=public_exponent,
+        key_size=key_size,
+    )
+    key_bytes = private_key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    )
+    return key_bytes.decode().strip()
+
+
+def get_certificate_extensions(
```

Comment:
```suggestion
def get_certificate_request_extensions(
```

Reply:
Aren't they? I mean the current name `get_certificate_extensions` would suggest that when implementing some feature I can use this function to see what extensions does the certificate have.
The docstring makes it clearer that that isn't the intended usage.
But the way this function is being used is that it is extracting extensions from the csr (cert request) and from parameters that come from the requirer's certificate request.
So at the end we are getting extensions from the request that we want to add to the certificate.

It's not a big deal, but the current name is misleading for me.
how about something like `get_certificate_extensions`?

Reply:
Sorry I had answered too quick there, I used your recommendation.